### ISSUE TITLE
Example not working with updated bootnodes

### ISSUE THREAD

**Issue by holgerd77** [ethereumjs/ethereumjs-devp2p#59]
Just injected the new bootnodes from https://github.com/ethereumjs/ethereumjs-vm/pull/718 manually to the current `master` version of the library.

The [peer-communication.ts](https://github.com/ethereumjs/ethereumjs-devp2p/blob/master/src/examples/peer-communication.ts) example is still not working, would want to wait with merging the bootnode update until we find the root cause here.

Running the example with the following command:

```shell
DEBUG='*devp2p*' node -r ts-node/register ./src/examples/peer-communication.ts
```

This leads to the following output:

```shell
  devp2p:rlpx refill connections.. queue size: 0, open slots: 25 +0ms
  devp2p:rlpx refill connections.. queue size: 0, open slots: 25 +10s
  devp2p:dpt call .refresh (0 peers in table) +10s
Total nodes in DPT: 0, open slots: 25, queue: 0 / 0
  devp2p:rlpx refill connections.. queue size: 0, open slots: 25 +11ms
  devp2p:rlpx refill connections.. queue size: 0, open slots: 25 +10s
  devp2p:rlpx refill connections.. queue size: 0, open slots: 25 +10s
```

So there are no new peers added during connection build up.

Normally the output should be much more diverse, there is the [debug](https://www.npmjs.com/package/debug) package used for this (thus the `DEBUG='*devp2p*'` appendix on the example command, there should be actually some more explicit explanation in the README on how to adopt here).

I tried to switch to `DEBUG=*`, output still remains limited. Normally there should be output from various loggers, e.g. [devp2p:rlpx:peer](https://github.com/ethereumjs/ethereumjs-devp2p/blob/master/src/rlpx/peer.ts#L12). Not sure, didn't give this much time to test.

This issue is for having a place to debug this and reporting on the progress.

**Comment by PhilippLgh** [comment]
tried it myself by pasting the bootnodes directly likes this:
(please note hardcoded chanid)
```
const Common = require('ethereumjs-common').default
const config = new Common('mainnet')
//const bootstrapNodes = config.bootstrapNodes()
const bootstrapNodes = [
  {
    
    "ip": "18.138.108.67",
    "port": "30303",
    "id": "d860a01f9722d78051619d1e2351aba3f43f943f6f00718d1b9baa4101932a1f5011f16bb2b1bb35db20d6fe28fa0bf09636d26a87d31de9ec6203eeedb1f666",
    "location": "ap-southeast-1-001",
    "comment": "bootnode-aws-ap-southeast-1-001"
  },
  ...
].map(node => ({...node, chainId: 1}))
```

 and I get a lot of output:
```
node -r ts-node/register ./src/examples/peer-communication.ts

Total nodes in DPT: 8, open slots: 25, queue: 0 / 0
Add peer: 5....657 gqdc/v1.2.0-stable-1528b791/linux-amd64/go1.12.3 (eth63) (total: 1)
Add peer: 51...0303 Geth/v1.9.12-stable-657603ab/linux-amd64/go1.13.4 (eth63) (total: 2)
Add peer: 1....62:30303 Parity-Ethereum/v2.5.13-stable-253ff3f-20191231/x86_64-linux-gnu/rustc1.40.0 (eth63) (total: 3)
Add peer: 3....2:30311 Geth/v1.9.6-unstable/linux-amd64/go1.11.13 (eth63) (total: 4)
Add peer: 3...67:30303 Geth/v1.9.10-stable-58cf5686/linux-amd64/go1.13.6 (eth63) (total: 5)
Remove peer: 1....62:30303 - peer disconnect, reason: undefined (undefined) (total: 4)
Add peer: 1...05:30303 Parity-Ethereum/v2.2.7-stable-b00a21f-20190115/x86_64-linux-gnu/rustc1.31.1 (eth63) (total: 5)
Peer error (3...2:30311): NetworkId mismatch: 01 / 4c6b7d38
Add peer: 3....46:30303 Parity-Ethereum/v2.5.13-stable-253ff3f-20191231/x86_64-linux-gnu/rustc1.40.0 (eth63) (total: 6)
Add peer: 13...89:30303 Parity-Ethereum/v2.5.13-stable-253ff3f-20191231/x86_64-linux-gnu/rustc1.40.0 (eth63) (total: 7)
Remove peer: 5....7:28657 - peer disconnect, reason: SUBPROTOCOL_ERROR (16) (total: 6)
Remove peer: 14...5:30303 - peer disconnect, reason: undefined (undefined) (total: 5)
Remove peer: 3...2:30311 - we disconnect, reason: SUBPROTOCOL_ERROR (16) (total: 4)
Remove peer: 13...89:30303 - peer disconnect, reason: TOO_MANY_PEERS (4) (total: 3)
Remove peer: 5...16:30303 - peer disconnect, reason: SUBPROTOCOL_ERROR (16) (total: 2)
```
my guess is that the chainId is either missing or bad in your tests.

**Comment by PhilippLgh** [comment]
will try to link the common module and do more manual testing and review today

**Comment by PhilippLgh** [comment]
ok just confirmed that this line https://github.com/ethereumjs/ethereumjs-devp2p/blob/master/src/examples/peer-communication.ts#L20
is causing the example to fail as the `chainId`is not appened to the json and not specified there either. 
checked the old config and `ropsten.json` was the only file that did specify a chainId. 
so we either:
1. remove the filter from example
2. write chainId in all .json files
3. let the common lib append the correct chainId to bootnodes (just as I didin the example)

**Comment by holgerd77** [comment]
Thanks for the analysis. 👍 The filter actually makes no sense since the `Common` instance only returns the bootnodes for the network set anyhow when calling `config.bootstrapNodes()`.

So we can safely go for 1. and just remove the filter here without loosing anything. Can you prepare a PR here?

Can confirm that this works now (like a charm). 😄

**Comment by PhilippLgh** [comment]
removed the filter here:
https://github.com/ethereumjs/ethereumjs-devp2p/pull/60

**Comment by holgerd77** [comment]
Closed by #67
